packages/tailwindcss/src/walk.ts

```diff
@@ -0,0 +1,462 @@
+const enum WalkKind {
+  Continue,
+  Skip,
+  Stop,
+  Replace,
+  ReplaceSkip,
+  ReplaceStop,
+}
+
+export const WalkAction = {
+  Continue: { kind: WalkKind.Continue } as const,
+  Skip: { kind: WalkKind.Skip } as const,
+  Stop: { kind: WalkKind.Stop } as const,
+  Replace: <T>(nodes: T | T[]) =>
+    ({ kind: WalkKind.Replace, nodes: Array.isArray(nodes) ? nodes : [nodes] }) as const,
+  ReplaceSkip: <T>(nodes: T | T[]) =>
+    ({ kind: WalkKind.ReplaceSkip, nodes: Array.isArray(nodes) ? nodes : [nodes] }) as const,
+  ReplaceStop: <T>(nodes: T | T[]) =>
+    ({ kind: WalkKind.ReplaceStop, nodes: Array.isArray(nodes) ? nodes : [nodes] }) as const,
```

Comment:
The only reason these are uppercase is because they represent enum values. We don't have ADTs in JS sadly, so we work around it by having a function that returns a specific object.

All of them have a unique `kind` discriminant.

We already had `Continue`, `Skip`, `Stop`, and `replaceWith`. I then added `Replace(…)`, `ReplaceSkip(…)` and `ReplaceStop(…)`.

Open for suggestions for better names here.